libraries/AP_Math/AP_Math_Common.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <limits>
+#include <math.h>
+
+
+/* 
+ * @brief: This magic functions swaps the value of the two input parameters
+ */
+template <class T> 
+void swap(T& a, T& b) {
```

Comment:
In AP_Compass_HMC5843 we are using std::swap(). Is there any problem with that that we should prefer our own version?